src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -800,7 +800,11 @@ public Task<HtmlString> RenderSectionAsync(string name, bool required)
             else if (required)
             {
                 // If the section is not found, and it is not optional, throw an error.
-                throw new InvalidOperationException(Resources.FormatSectionNotDefined(sectionName, Path));
+                var message = Resources.FormatSectionNotDefined(
+                    ViewContext.ExecutingFilePath,
```

Comment:
Not that important but at least FYI: `ExecutingFilePath` and `Path` are identical here. Executing the layout page.